### PR TITLE
Update Introduction in draft-bradleylundberg-cfrg-arkg.md

### PR DESCRIPTION
**Pull request by AltmannPeter**
Restructured the text a bit so that the problem statement is earlier, that the procedures required are enumerated, and to position ARKG in relation to KBSS.

Changed some terms. 
- the seed is not private public. It is private, shared secret. If it was public, then any entity could derive the same keys, rendering the entire proposal moot.
- reduced the number of terms used to describe similar concepts. For instance, 'intended private key holder' instead of delegating party, subordinate entity instead of public key consumer etc.
- Changed the order of how the text uses examples. Examples are now used to illustrate the benefit of a certain ARKG property.

Overall changes to readability and clarity (I hope).

### CODE REVIEW

File: draft-bradleylundberg-cfrg-arkg.md [hunk 1]
```diff
@@ -131,75 +131,39 @@ We expect that additional instances will be defined in the future.
...
-The algorithm consists of three procedures:
-(1) the _delegating party_ generates a _seed pair_ and emits the _public seed_ to a _subordinate party_,
-(2) the subordinate party uses the public seed to generate a public key and a _key handle_ on behalf of the delegating party, and
-(3) the delegating party uses the key handle and the _private seed_
-to derive the private key corresponding to the public key generated by procedure (2).
-Procedure (1) is performed once, and procedures (2) and (3) may be repeated any number of times with the same seed pair.
-The required cryptographic primitives are a public key blinding scheme, a key encapsulation mechanism (KEM),
-a key derivation function (KDF) and a message authentication code (MAC) scheme.
-Both conventional primitives and quantum-resistant alternatives exist that meet these requirements. [Wilson]
+Asynchronous Remote Key Generation (ARKG) introduces a mechanism to generate public keys without access to the corresponding private keys, and without anyone other than the intended private key holder being able to generate these private keys.
```
Comment (emlun) [inline draft-bradleylundberg-cfrg-arkg.md]: >and without anyone other than the intended private key holder being able to generate these private keys

This seems redundant, already covered by the first half?

File: draft-bradleylundberg-cfrg-arkg.md [hunk 2]
```diff
@@ -131,75 +131,39 @@ We expect that additional instances will be defined in the future.
...
+Such a mechanism is pivotal for many scenarios where private key holders cannot engage in the generation of new key pairs whenever a public key is needed.
```
Comment (emlun) [inline draft-bradleylundberg-cfrg-arkg.md]: Replacing the "delegating party" term would have to be done throughout the document as a whole, not just the introduction. I'm reverting to using the "delegating party" term for now.

File: draft-bradleylundberg-cfrg-arkg.md [hunk 3]
```diff
@@ -131,75 +131,39 @@ We expect that additional instances will be defined in the future.
...
+This may occur when the hardware security device that protects the private keys is unavailable at the time of generation. ARKG is delineated through three primary procedures:
 
+* __Initialization__: The intended private key holder generates a _seed pair_, disclosing the _shared secret seed_ to a subordinate entity while securely retaining the _private seed_.
```
Comment (emlun) [inline draft-bradleylundberg-cfrg-arkg.md]: I like the term "shared seed" (it does not always need to be secret, depending on application) as opposed to "public seed", but this also needs to be changed throughout the whole document. Reverting to "public seed" for now, but we can change to "shared seed" in a separate PR.

File: draft-bradleylundberg-cfrg-arkg.md [hunk 4]
```diff
@@ -131,75 +131,39 @@ We expect that additional instances will be defined in the future.
...
-
-- Enhanced forward secrecy for encrypted messaging.
-  For example, [section 8.5.4 of RFC 9052][rfc9052-direct-key-agreement] defines COSE representations for encrypted messages and notes that
+- __Single-use asymmetric keys__:
+  Envisioned for the European Union's digital identity framework, ARKG facilitates the generation of unique key pairs for individual authentication processes.
+  Unique single-use asymmetric keys prevent colluding verifiers from using the public key material as a correlation handle.
+  In online usage, ARKG enables public key generation on demand. This is useful when creating single-use certificates, single-use proof of possession keys to be included in (qualified) electronic attestations of attributes or personal identification data attestations.
+  In offline usage, ARKG enables pre-generation of single-use keys.
+  One candidate implementation for single-use keys is the W3C Web Authentication API [WebAuthn] (WebAuthn), which presently requires private key holder engagement whenever a WebAuthn operation is invoked.
```
Comment (emlun) [inline draft-bradleylundberg-cfrg-arkg.md]: >which presently requires private key holder engagement

Not just presently, it's unlikely that this restriction will ever be relaxed in WebAuthn.

File: draft-bradleylundberg-cfrg-arkg.md [hunk 5]
```diff
@@ -131,75 +131,39 @@ We expect that additional instances will be defined in the future.
...
+- __Enhanced forward secrecy__:
+  The use of ARKG facilitates forward secrecy in certain contexts. For instance, [section 8.5.4 of RFC 9052][rfc9052-direct-key-agreement] defines COSE representations for encrypted messages and notes that
   "Since COSE is designed for a store-and-forward environment rather than an online environment,
-  \[...\] forward secrecy (see [RFC4949]) is not achievable. A static key will always be used for the receiver of the COSE object."
```
Comment (emlun) [inline draft-bradleylundberg-cfrg-arkg.md:179]: These backslashes are needed so the RFC "compiler" doesn't interpret this as a reference.

File: draft-bradleylundberg-cfrg-arkg.md [hunk 6]
```diff
@@ -131,75 +131,39 @@ We expect that additional instances will be defined in the future.
...
+Such a mechanism is pivotal for many scenarios where private key holders cannot engage in the generation of new key pairs whenever a public key is needed.
+This may occur when the hardware security device that protects the private keys is unavailable at the time of generation. ARKG is delineated through three primary procedures:
 
+* __Initialization__: The intended private key holder generates a _seed pair_, disclosing the _shared secret seed_ to a subordinate entity while securely retaining the _private seed_.
+* __Public key generation__: With the shared seed, the subordinate entity autonomously generates any number of new public keys alongside a unique _key handle_ for each invocation.
+* __Private key derivation__: The intended private key holder jointly employs the key handle and the private seed to derive the private keys corresponding to the public keys generated by procedure two.
+
+One application is key blinding where the private key is a function of a long-term private key and a freshly generated blinding key, and correspondingly where the public key is a function of the long-term public key and the same blinding key.
```
Comment (emlun) [inline draft-bradleylundberg-cfrg-arkg.md]: I think this is mostly covered by the examples below?

File: draft-bradleylundberg-cfrg-arkg.md [hunk 7]
```diff
@@ -131,75 +131,39 @@ We expect that additional instances will be defined in the future.
...
-  This makes it difficult for users to set up a backup authenticator,
-  because each time a key pair is created for the primary authenticator,
-  another key pair also needs to be created for the backup authenticator, which may be stored in a safe but inconvenient location.
 
-  ARKG could enable the primary authenticator to also generate a public key for a paired backup authenticator
-  whenever it generates a key pair for itself,
-  allowing the user to set up the pairing once
-  and then leave the backup authenticator in safe storage until the primary authenticator is lost.
+Notably, ARKG can be built entirely using commonly deployed cryptographic primitives. The required primitives are a key encapsulation mechanism (KEM),
+a key derivation function (KDF), and a message authentication code (MAC) scheme. Both conventional primitives and quantum-resistant alternatives exist that meet these requirements. [Wilson]
```
Comment (emlun) [inline draft-bradleylundberg-cfrg-arkg.md]: Why remove the key blinding scheme from the list of primitives?

File: draft-bradleylundberg-cfrg-arkg.md [hunk 8]
```diff
@@ -131,75 +131,39 @@ We expect that additional instances will be defined in the future.
...
-  which requires a user gesture whenever a WebAuthn operation is invoked.
-  A WebAuthn-based implementation of the proposed digital identity system
-  could use ARKG to pre-emptively generate key pairs for offline use without the need to prompt for a user gesture for each key pair generated.
-
-- Enhanced forward secrecy for encrypted messaging.
-  For example, [section 8.5.4 of RFC 9052][rfc9052-direct-key-agreement] defines COSE representations for encrypted messages and notes that
+- __Single-use asymmetric keys__:
+  Envisioned for the European Union's digital identity framework, ARKG facilitates the generation of unique key pairs for individual authentication processes.
+  Unique single-use asymmetric keys prevent colluding verifiers from using the public key material as a correlation handle.
+  In online usage, ARKG enables public key generation on demand. This is useful when creating single-use certificates, single-use proof of possession keys to be included in (qualified) electronic attestations of attributes or personal identification data attestations.
```
Comment (emlun) [inline draft-bradleylundberg-cfrg-arkg.md]: The online use case isn't a very strong motivation, as that can almost as easily be done with conventional key generation. The offline use case is much stronger in my opinion: pre-generating keys in large batches at once without having to call out to the secure element (possibly invoking a user gesture) for each key.

**Review by emlun — COMMENTED** [review]
There's a lot I like here! I ended up rewriting and restructuring a lot of it, though, to cut down on some redundancy and try to say the same things in fewer words.

I also moved the "three procedures" to after the examples, I think the section flows a little better now with ending on the declaration that quantum-resistant instantiations exist and not ending abruptly after a bullet-point list.

I hope I've preserved the spirit of what you wanted to accomplish, do you agree?

File: draft-bradleylundberg-cfrg-arkg.md (hunk 8 above)
Comment (sander) [reply draft-bradleylundberg-cfrg-arkg.md]: The online use case for single-use, just-in-time PID/(Q)EAA still looks valid:

- Creating a brand new ECDSA key in the mobile SCE does not provide the issuer any assurance that it is bound to the same device as the PID that was presented to them. With ARKG, the issuer can have this assurance.
- Creating a brand new ECDSA key in the mobile SCE requires additional key management, which brings the latency of key generation and in the Android case also the problem of key deletion. With ARKG, the SCE does not need to do any additional key generation or deletion.

**Pull request merged**